terminus-ui/drawer/src/drawer.component.ts

```diff
@@ -0,0 +1,343 @@
+    // fires twice on some browsers.
+    this.animationEnd.pipe(
+      distinctUntilChanged((x, y) => x.fromState === y.fromState && x.toState === y.toState),
+      untilComponentDestroyed(this),
+    ).subscribe((event: AnimationEvent) => {
+      const { fromState, toState } = event;
+
+      if ((toState.indexOf('open') === 0 && fromState === 'void')
+        || (toState === 'void' && fromState.indexOf('open') === 0)
+      ) {
+        this.expandedChange.emit(this.isExpanded);
+      }
+    });
+    this.renderer.setStyle(this.elementRef.nativeElement, 'width', this._expandedSize);
+  }
+
+  public ngAfterContentChecked() {
+    // Enable the animations after the lifecycle hooks have run, in order to avoid animating
+    // drawers that are open by default.
+    if (this.platform.isBrowser) {
+      this.enableAnimations = true;
+    }
+  }
+
+  public ngOnDestroy() {
+    this.modeChanged.complete();
+    this.destroyed.next();
+    this.destroyed.complete();
+  }
+
+  /**
+   * Expand the drawer.
+   *
+   * @return Promise<TsDrawerToggleResult>
+   */
+  public expand(): Promise<TsDrawerToggleResult> {
+    return this.toggle(true);
+  }
+
+  /**
+   * Collapse the drawer.
+   *
+   * @return Promise<TsDrawerToggleResult>
+   */
+  public collapse(): Promise<TsDrawerToggleResult> {
+    this.elementRef.nativeElement.classList.add('ts-drawer--collapsed');
+    return this.toggle(false);
+  }
+
+  /**
+   * Toggle this drawer.
+   *
+   * @param isOpen - whether the drawer should be open.
+   * @return  Promise<TsDrawerToggleResult>
+   */
+  public toggle(isOpen: boolean = !this.isExpanded): Promise<TsDrawerToggleResult> {
+    this._isExpanded = isOpen;
+    if (isOpen) {
+      this.animationState = this.enableAnimations ? 'open' : 'open-instant';
+      // this.elementRef.nativeElement.focus();
+    } else {
+      this.animationState = 'void';
+    }
+
+    return new Promise<TsDrawerToggleResult>(resolve => {
+      this.expandedChange.pipe(take(1)).subscribe(open => resolve(open ? 'open' : 'close'));
+    });
+  }
+
+  // We have to use a `HostListener` here in order to support both Ivy and ViewEngine.
+  // In Ivy the `host` bindings will be merged when this class is extended, whereas in
+  // ViewEngine they're overwritten.
+  // TODO: we move this back into `host` once Ivy is turned on by default.
+  // tslint:disable-next-line:no-host-decorator-in-concrete
+  @HostListener('@transform.start', ['$event'])
+  public animationStartListener(event: AnimationEvent) {
+    this.animationStarted.next(event);
+  }
+  //
+  // We have to use a `HostListener` here in order to support both Ivy and ViewEngine.
+  // In Ivy the `host` bindings will be merged when this class is extended, whereas in
+  // ViewEngine they're overwritten.
+  // TODO: move this back into `host` once Ivy is turned on by default.
+  // tslint:disable-next-line:no-host-decorator-in-concrete
```

Comment:
I'm unfamiliar with `no-host-decorator-in-concrete`.. is that a custom lint rule?